src/uucore/src/lib/lib.rs

```diff
@@ -179,6 +183,28 @@ pub fn execution_phrase() -> &'static str {
     &EXECUTION_PHRASE
 }
 
+struct SimpleLogSink {
+    file: Option<OwnedFileDescriptorOrHandle>,
+}
+
+pub fn uu_log_file() -> Box<dyn Write> {
+    static SINK: Lazy<SimpleLogSink> = Lazy::<SimpleLogSink>::new(|| SimpleLogSink {
+        file: std::env::var_os(UUTILS_LOG_FILE_ENV_NAME)
+            .and_then(|path| File::create(path).ok())
+            .and_then(|f| OwnedFileDescriptorOrHandle::from(f).ok()),
+    });
+    SINK.file
+        .as_ref()
+        .and_then(|f| f.try_clone().ok())
+        .map(|fx| Box::new(fx.into_file()) as Box<dyn Write>)
+        .unwrap_or(Box::new(std::io::sink()))
+}
+
+#[macro_export]
+macro_rules! uu_log {
+    ($($arg:tt)*) => { let _ = writeln!(uu_log_file(), $($arg)*); };
```

Comment:
This will still cause all arguments to be evaluated and formatted. If that wouldn't be the case, it might be useful as a `eprintln!()` replacement, i.e.: Not used anywhere by default, but it's easily available while debugging.

Reply:
> This will still cause all arguments to be evaluated and formatted. If that wouldn't be the case, it might be useful as a `eprintln!()` replacement

Not sure how a eprintln!() would not cause all arguments to be evaluated!?
As described in the description of the PR, the calls to `uu_log!()` would be removed when debugging is done. So their performance impact is of second priority.

Reply:
I think its a minor change, but I adapted the code anyway such that the arguments are not evaluated if the logging is not enabled.